.github/workflows/release_github.yaml

```diff
@@ -0,0 +1,113 @@
+
+env:
+  BRANCH: ${{ github.ref_name }}
+  VERSION: 0.0.0
+
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+      - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  checks:
+    name: Check build condition for spark libs
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version:
+          - '3.9'
+          - '3.10'
+    steps:
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.BRANCH }}
+          fetch-depth: 0
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: ${{matrix.python-version}}
+          architecture: x64
+      - id: code-checks
+        name: Make sure the code follows conventions and tests correctly
+        run: |
+          make checks
+
+  autorelease:
+    name: Release the package on github
+    needs: checks
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      fail-fast: true
+    steps:
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.BRANCH }}
+          fetch-depth: 0
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: '3.10'
+          architecture: x64
+      - id: build-package
+        name: Build package
+        run: |
+          make checks
+          poetry build
+      - name: Add version to environment vars
+        run: |
+          echo "PROJECT_VERSION=${{ env.VERSION }}" >> $GITHUB_ENV
+      - name: Check if tag version matches project version
+        run: |
+          TAG=$(git describe HEAD --tags --abbrev=0)
+          echo $TAG
+          echo $PROJECT_VERSION
+          if [[ "$TAG" != "v$PROJECT_VERSION" ]]; then exit 1; fi
+      - name: Create Github Release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          body_path: ".github/RELEASE-TEMPLATE.md"
+          files: |
+            dist/spark_client-${{env.PROJECT_VERSION}}-py3-none-any.whl
+            dist/spark_client-${{env.PROJECT_VERSION}}.tar.gz
+
+  test:
+    name: Test Release
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    env:
+      URL_BASE: "https://github.com/canonical/spark-k8s-toolkit-py/releases/download"
+      PKG_NAME: spark_client
+    needs:
+      - autorelease
+      - checks
+    steps:
+      - id: check-tar-gz
+        name: Check tar.gz package
+        run: |
+          # check if release is now published and available
+          echo "Checking latest available Spark package release v${{env.VERSION}}."
+          STATUSCODE=$(curl --silent --head ${{env.URL_BASE}}/v${{env.VERSION}}/${{env.PKG_NAME}}-${{env.VERSION}}.tar.gz | head -n 1 | cut -d' ' -f2)
```

Comment:
Let me know if there may be a more elegant way for these variables. If I gathered correctly, github workflow YAML-s don't allow for redundant references within `env` (i.e. `env: X_${{env.Y}}`), thus this seemed the best I could do -- but I may be wrong.

Reply:
That's fine to concatenate multiple variable. But rather, it could be that some of these variables do not need to be declared, but rather we can either parse them from repo files (e.g. pyproject.toml) or from the github context (e.g. repo URL)

Reply:
I'm happy to do that, it's been expressed on my side a number of times how I'm missing that feature.
My plan was not to load this PR further with that, but if preferred, I'm glad to do it here.

Reply:
(As I'm worried that it's non-trivial to get typically the `VERSION` value from `pyproject.toml` -- if it is I'm more than happy to remove the redundancy on the spot)

Reply:
Let's sync about this after daily sync (my morning is packed now), to decide the way forward whether address this in the PR or in a separate item if the scope goes beyond the PR

Reply:
Sure. @welpaolo is there a chance you can join? (We've discussed this particular matter with him just a few mins ago.)

Reply:
I did what I could... It's not very pretty, I couldn't find a neat way to do this. Potential better solution is welcome.